pkg/server/sandbox_run.go

```diff
@@ -124,7 +124,28 @@ func (c *criContainerdService) RunPodSandbox(ctx context.Context, r *runtime.Run
 	}
 
 	// Start sandbox container.
-	spec := c.generateSandboxContainerSpec(id, config)
+	hostNetwork := config.GetLinux().GetSecurityContext().GetNamespaceOptions().GetHostNetwork()
+	var netNSPath string
+	if hostNetwork {
+		netNSPath, err = hostNetNsPath()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get host netns path: %v", err)
+		}
+	} else {
+		// Create the sandbox network namespace
+		if err = meta.NetNSCreate(); err != nil {
```

Comment:
We should be aware of that, if `cri-containerd` crashes after we create the namespace, but before we checkpoint the namespace path, we'll not be able to cleanup the namespace.

Add a TODO here to handle that case in the future.